### ISSUE TITLE
tex labels in MultiIndex columns

### ISSUE THREAD

**Issue by williamjameshandley** [williamjameshandley/anesthetic#214]
As @lukashergt has pointed out, with the advent of removing limits (#200, #212), `tex` is the only remaining array we are carrying around as an attribute in `Samples` objects. As @lukashergt again pointed out, a more consistent approach with anesthetic would be to include it as a MultiIndex, for example:
```python
from anesthetic import NestedSamples 
from pandas import MultiIndex                                                                                                                              
                                                                                                
samples = NestedSamples(root='./tests/example_data/pc')                                         
tex = [samples.tex[p] for p in samples.columns]                                                 
columns = MultiIndex.from_arrays([samples.columns, tex], names=[None, 'tex'])     
samples.columns=columns            
print(samples.iloc[:,:5])                                                                       
```
which in a terminal produces:
```
                          x0        x1        x2        x3        x4                            
tex                    $x_0$     $x_1$     $x_2$     $x_3$     $x_4$                            
#    weights                                                                                    
0    0.000000e+00   0.839566 -0.927932  0.174772  0.389135  3.904285                            
1    1.088407e-308  0.711607 -0.976572  0.359978  0.729991  2.343259                            
2    4.830017e-301 -0.703718  0.963659  0.478370  0.497391  3.278945                            
3    6.951167e-262  0.980678 -0.566206  0.663078  0.263322  2.041302                            
4    7.661544e-241  0.727711 -0.764936  0.517379  0.132068  3.433719                            
...                      ...       ...       ...       ...       ...                            
1495 2.273727e-01  -0.025695 -0.027214  0.014068  0.923650  3.960524                            
1496 2.278716e-01   0.006843  0.018013  0.015020  0.863035  3.987410                            
1497 2.314056e-01   0.023063  0.025451  0.002778  0.566811  3.773360                            
1498 2.333173e-01   0.017399  0.005777  0.012071  0.825495  3.978778                            
1499 2.381040e-01  -0.013951 -0.025874  0.003703  0.833060  3.886403                            
                                                                                                
[1500 rows x 5 columns]                                                                         
```
This can be rendered as latex in a jupyter notebook.

To do:
- write helper functions for adjusting tex labels, since adjusting a single index directly is not possible in pandas.
- choose name -- is 'tex' correct? 'label' may be better/more pandas consistent, since labels do not need to be latex. We may also want to name the first level as e.g. 'name', or leave it blank.



The main issue which arises is with long tex labels, for example ```print(samples)``` gives
```
name                      x0        x1  ...                      logL_birth             nlive   
tex                    $x_0$     $x_1$  ... $\ln\mathcal{L}_\mathrm{birth}$ $n_\mathrm{live}$   
#    weights                            ...                                                     
0    0.000000e+00   0.839566 -0.927932  ...                            -inf               125   
1    1.088407e-308  0.711607 -0.976572  ...                            -inf               125   
2    4.830017e-301 -0.703718  0.963659  ...                            -inf               125   
3    6.951167e-262  0.980678 -0.566206  ...                            -inf               125   
4    7.661544e-241  0.727711 -0.764936  ...                            -inf               125   
...                      ...       ...  ...                             ...               ...   
1495 2.273727e-01  -0.025695 -0.027214  ...                        4.239939                 5   
1496 2.278716e-01   0.006843  0.018013  ...                        5.055034                 4   
1497 2.314056e-01   0.023063  0.025451  ...                        5.196540                 3   
1498 2.333173e-01   0.017399  0.005777  ...                        3.638035                 2   
1499 2.381040e-01  -0.013951 -0.025874  ...                        4.860346                 1   
                                                                                                
[1500 rows x 8 columns]                                                                         
```

In principle one should be able to control this with `pandas.options.display.max_colwidth=14`, but this only limits the widths of columns, not column headers. This is a bug in pandas which may one day be fixed https://github.com/pandas-dev/pandas/issues/7059, https://github.com/pandas-dev/pandas/issues/16911.

**Comment by williamjameshandley** [comment]
Having had a go at implementing this, there are issues which immediately arise. If we have a `MultiIndex` column, certain things become less intuitive. 

For example: `samples.logL <= samples.logL_birth` no longer works, since `samples.logL` returns a single column `NestedSamples` object with column name `'$\log\mathcal{L}$'`, while `samples.logL_birth` returns one with column name `'$\log\mathcal{L}_\mathrm{birth}$'`, which are deemed incompatible, and raise a `ValueError: Can only compare identically-labeled DataFrame objects`.

I'm not sure this issue is surmountable -- any other ideas for a better treatment of the tex variable @lukashergt.

**Comment by lukashergt** [comment]
Ah, what a shame. I would have liked the tex label in the printed table...

Would column metadata be something worth thinking about?
```python
samples.x0.label = r'$x_0$'
```

**Comment by williamjameshandley** [comment]
I think that could actually work quite well (famous last words). Now that I've dug around in the printing, there may yet be a way to get printing to still have an apparent multiindex column if that would be desired.

**Comment by lukashergt** [comment]
> Having had a go at implementing this, there are issues which immediately arise. If we have a `MultiIndex` column, certain things become less intuitive.
> 
> For example: `samples.logL <= samples.logL_birth` no longer works, since `samples.logL` returns a single column `NestedSamples` object with column name `'$\log\mathcal{L}$'`, while `samples.logL_birth` returns one with column name `'$\log\mathcal{L}_\mathrm{birth}$'`, which are deemed incompatible, and raise a `ValueError: Can only compare identically-labeled DataFrame objects`.
> 
> I'm not sure this issue is surmountable -- any other ideas for a better treatment of the tex variable @lukashergt.

In light of recent difficulties of keeping the tex labels along for the ride in #215 (see [this comment](https://github.com/williamjameshandley/anesthetic/pull/215#issuecomment-1216481434) in particular), are the MultiIndex problems that bad?

We can easily solve `samples.logL <= samples.logL_birth` with `samples.logL.to_numpy() <= samples.logL_birth.to_numpy()`... I agree that this is slightly annoying for the user, but it comes with clear error messages and has clear pathways to solutions with `to_numpy()` and `droplevel('tex')`.

**Comment by lukashergt** [comment]
This got resolved with #215.
